strapi/strapi
# Chore/user level data

### What does it do?

This PR introduces refactored analytics that are now sending user-level context to the analytics server. Previously it was project-level context.

***Important***: users are still anonymous, analytics server only receives a hash of a user.

Changes that were introduced:
- All server and admin metrics calls now pass `adminUserId` property to their respective sender functions.
  - For admin metrics calls `adminUserId` is added at the sender function since admin user will be the same for the Admin Panel session
  - For server metrics calls, `adminUserId` is passed to every `strapi.telemetry.send()` call as the first argument. If the event supposed to be anonymous, empty string ( '' ) is passed in. 
- All generators events are by nature anonymous, but they still send an empty string as `adminUserId`
- Project uuid is now part of the `properties` object on all calls and has been renamed to `projectId`
- Tests were refactored to accommodate all these changes

### Why is it needed?

This is part of our initiative to migrate to user-level data analytics

### How to test it?

You can run the getstarted application and route all analytics events to your local server. You will be able to see the payload of the event.

***Apologies for a big PR, but there is pretty much no way to make it smaller as all these changes have to be merged at the same time.***

***Please don't merge this PR. I need it approved, but it will only be merged AFTER changes to the tracker are finalized and merged as well***
